app/assets/javascripts/miq_application.js

```diff
@@ -320,7 +320,10 @@ function miqCheckForChanges() {
     }
   }
   // use default browser reaction for onclick
-  return true;
+  if(obj && (href = obj.getAttribute("href")))
```

Comment:
@AparnaKarve That makes sense, it already knows it's changing state so it starts unloading.. But we could probably catch `$stateChangeStart` and do the `window.location.href = ...` in there (the handler does get passed `toState` and `toParams`), which should happen before the state starts unloading. 

Though it may not be a good idea if we ever want to use the `onExit` handler on a state. (But we could always emulate that one via `$scope.$on('$destroy', ...)`, so not a huge loss.)